repose-aggregator/src/docs/asciidoc/architecture/performance.adoc

```diff
@@ -0,0 +1,336 @@
+The name of the new directory should describe the feature under test.
+.. Add http://docs.ansible.com/ansible/intro_inventory.html#splitting-out-host-and-group-specific-data[group variables] in a `group_vars/` directory under your new inventory directory.
+Group variables should generally be placed in a file named `all` to make them usable by all host groups.
++
+[NOTE]
+====
+Certain variables (e.g., `gatling.test.name`) must be defined.
+====
+.. Add http://docs.ansible.com/ansible/intro_inventory.html#hosts-and-groups[a `hosts` file] under your new inventory directory.
++
+[NOTE]
+====
+Only `localhost` need be defined in the `hosts` file -- all other hosts will be added dynamically as the Playbook runs.
+====
+. Add Repose configuration files.
+.. Create new directories under `repose-aggregator/tests/performance-tests/roles/repose/` with the same names and hierarchy as the previously created inventory directory.
+Template files should be placed under the `templates/` directory while raw configuration files should be placed under the `files/config/` directory.
+. Create a http://gatling.io/docs/current/general/concepts/#simulation[Gatling simulation].
+.. Create new directories under `repose-aggregator/tests/performance-tests/roles/gatling/` with the same names and hierarchy as the previously created inventory directory.
+.. Write a Gatling simulation.
+This code will be responsible for generating test traffic and, ultimately, performance test data.
+. If necessary, add mock services under `repose-aggregator/tests/performance-tests/roles/repose/mocks/`.
+
+[NOTE]
+====
+The inventory directory created in the steps above is not strictly necessary.
+While it is the conventional way of defining a new test, Ansible can be invoked using any directory as the inventory directory.
+In fact, the directory that we have created will only be needed when we invoke the Ansible Playbook.
+The command in "How To Run Performance Tests Using Vagrant" shows the Ansible Playbook being invoked with the "-i" option, which allows a user to specify the inventory directory.
+====
+
+[NOTE]
+====
+Ansible Playbooks support Jinja2 templates.
+These templates allow users to add dynamic content to files.
+Most if not all of the configuration files that were referred to above will be Jinja2 files, distinguishable by the `.j2` file extension.
+====
+
+=== Running a Test Automatically
+
+A Jenkins job is triggered any time that code on the master branch of the https://github.com/rackerlabs/repose[Repose Git Repository] changes.
+The Jenkins job will run the performance tests automatically, and publish the results.
+<<Design>> describes the full automated process in more detail.
+
+=== Running a Test Manually
+
+[IMPORTANT]
+.TODO
+====
+* Add the link to the Puppet manifest.
+====
+
+. Set up your environment.
++
+[NOTE]
+====
+We use of Puppet to manage our performance test environment.
+The details of our environment can be found in our [Puppet manifest].
```

Comment:
Yes. Since the Puppet manifest does not yet exist, I left a TODO in the document.